internal/group.go

```diff
@@ -332,8 +330,8 @@ func (g *group) run(specs Specs) {
 					switch t := multipleExecutionDecorator; {
```

Comment:
rather than tracking `multipleExecutionDecorator` and doing a type switch could these be `if g.suite.currentSpecReport.MaxFlakeAttampts > 0`  and `else if g.suite.currentSpecReport.MaxMustPassRepeatedly > 0`?